applications/IgaApplication/custom_elements/shell_3p_element.h

```diff
@@ -0,0 +1,460 @@
+#if !defined(KRATOS_SHELL_3P_ELEMENT_H_INCLUDED )
+#define  KRATOS_SHELL_3P_ELEMENT_H_INCLUDED
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "utilities/math_utils.h"
+
+// Application includes
+#include "iga_application_variables.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+/// Short class definition.
+/** Kirchhoff-Love Shell. Optimized for Isogeometric Analysis by Kiendl et al. .
+*/
+class Shell3pElement
+    : public Element
+{
+protected:
+
+    /// Internal variables used for metric transformation
+    struct KinematicVariables
+    {
+        // covariant metric
+        array_1d<double, 3> a_ab_covariant;
+        array_1d<double, 3> b_ab_covariant;
+
+        //base vector 1
+        array_1d<double, 3> a1;
+        //base vector 2
+        array_1d<double, 3> a2;
+        //base vector 3 normalized
+        array_1d<double, 3> a3;
+        //not-normalized base vector 3
+        array_1d<double, 3> a3_tilde;
+
+        //differential area
+        double dA;
+
+        /**
+        * The default constructor
+        * @param Dimension: The size of working space dimension
+        */
+        KinematicVariables(SizeType Dimension)
+        {
+            a_ab_covariant = ZeroVector(Dimension);
+            b_ab_covariant = ZeroVector(Dimension);
+
+            a1 = ZeroVector(Dimension);
+            a2 = ZeroVector(Dimension);
+            a3 = ZeroVector(Dimension);
+
+            a3_tilde = ZeroVector(Dimension);
```

Comment:
I think all those `ZeroVector` assignments should be `noalias`

Reply:
Good point. Thanks!